shopinvader_product_seasonality/components/seasonal_config_line_event_listener.py

```diff
@@ -0,0 +1,29 @@
+# Copyright 2021 Camptocamp SA
+# @author: Simone Orsi <simone.orsi@camptocamp.com>
+# License AGPL-3.0 or later (http://www.gnu.org/licenses/agpl).
+from odoo.addons.component.core import Component
+from odoo.addons.component_event import skip_if
+
+
+class SeasonalConfigLineEventListener(Component):
+    _name = "seasonal.config.line.event.listener"
+    _inherit = "base.event.listener"
+
+    _apply_on = ["seasonal.config.line"]
+
+    @skip_if(lambda self, record, **kw: self._check_product_bindings(record, **kw))
+    def on_record_create(self, record, fields=None):
+        self._create_config_line_bindings_if_missing(record)
+
+    def _check_product_bindings(self, record, **kw):
+        return not any(
+            (
+                record.product_id.shopinvader_bind_ids,
+                record.product_template_id.shopinvader_bind_ids,
+            )
+        )
```

Comment:
```suggestion
    @skip_if(lambda self, record, **kw: self._check_product_bindings(record, **kw))
    def on_record_create(self, record, fields=None):
        self._create_config_line_bindings_if_missing(record)

    def _check_product_bindings(self, record, **kw):
        return any(
            (
                record.product_id.shopinvader_bind_ids,
                record.product_template_id.shopinvader_bind_ids,
            )
        )
```

Maybe I'm still wrong, but That I read is:
"Skip binding creation if we don't have an existing binding."

I would have:
"Skip binding creation if we have an existing binding."

Reply:
no, you are mixing the bindings: we have bindings on the products and bindings for the config lines.
We DON'T want to create config line bindings if the related product has no binding. :wink:

Reply:
Hum: immediately more clear with the explanation :)
A little comment maybe to the future? ;)